From d329e790180bfc6a0d29a5a3560485d20b88bba2 Mon Sep 17 00:00:00 2001
From: Kai Luo <gluokai@gmail.com>
Date: Thu, 21 Mar 2024 22:55:16 +0800
Subject: [PATCH 08/10] Backport jitlink ppc64 (#13850)

* Backport JITLink ppc64 backend to LLVM-16

* Backport ELF part

* Backport PCREL relocations

* Use jitlink::Section::blocks::empty instead

* Backport TLS PCREL relocation

* [llvm-project] Synchronize with LLVM monorepo fork

---------

Co-authored-by: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>
---
 lib/Interpreter/IncrementalJIT.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Interpreter/IncrementalJIT.cpp b/lib/Interpreter/IncrementalJIT.cpp
index fd9930c4..e8f11163 100644
--- a/lib/Interpreter/IncrementalJIT.cpp
+++ b/lib/Interpreter/IncrementalJIT.cpp
@@ -442,7 +442,8 @@ static bool UseJITLink(const Triple& TT) {
   // LLJITBuilderState::prepareForConstruction.
   if (TT.getArch() == Triple::riscv64 ||
       (TT.isOSBinFormatMachO() &&
-       (TT.getArch() == Triple::aarch64 || TT.getArch() == Triple::x86_64))) {
+       (TT.getArch() == Triple::aarch64 || TT.getArch() == Triple::x86_64)) ||
+      (TT.isOSBinFormatELF() && TT.getArch() == Triple::ppc64le)) {
     jitLink = true;
   }
   // Finally, honor the user's choice by setting an environment variable.
